polygeist/include/mlir/Dialect/Polygeist/Utils/TransformUtils.h

```diff
@@ -49,139 +39,151 @@ void fully2ComposeAffineMapAndOperands(PatternRewriter &rewriter,
                                        DominanceInfo &DI);
 bool isValidIndex(Value val);
 
+//===----------------------------------------------------------------------===//
+// Interface classes
+//===----------------------------------------------------------------------===//
+class IfThenElseBodyInterface {
```

Comment:
What's the purpose of `IfThenElseBodyInterface`? It only has two pure virtual functions.

Reply:
It is an interface class (all pure virtual functions) and is used by `LoopVersionBuilder` and `LoopGuardBuilder` to indicate that their derived classes needs to implement the member functions declared pure virtual.

Reply:
Personally don't find it very useful, as `LoopVersionBuilder` and `LoopGuardBuilder` could decide on how themselves should/want to be implemented. Is ok if you want to keep it.